Rego/Utils/DefenderUtils.rego

```diff
@@ -115,4 +119,45 @@ SensitiveAccounts(SensitiveAccountsSetting, SensitiveAccountsConfig) := true if
         SensitiveAccountsSetting.ExcludedDomains == SensitiveAccountsConfig.ExcludedDomains
     ]
     count([x | x := Mismatches[_]; x == false]) == 0
+}
+
+##############################################
+# Inpersonation protection support functions #
+##############################################
+
+InpersonationProtectionSetting(Policies, IdentityString) := Policy if {
+    Policy := [
+        Policy | Policy := Policies[_];
+        regex.match(IdentityString, Policy.Identity) == true;
+        Policy.Enabled == true;
+        Policy.EnableTargetedUserProtection == true
+    ]
+} else := set()
+
+InpersonationProtectionConfig(PolicyID) := IncludedUsers if {
+    SensitiveUsers := input.scuba_config.Defender[PolicyID].SensitiveAccounts
+    IncludedUsers := { trim_space(x) | some x in SensitiveUsers.IncludedUsers; x != null }
+} else := set()
+
+InpersonationProtection(Policies, IdentityString, IncludedUsers) := {
+    "Result" : true,
+    "Policy" : {
+        "Name" : Policy[0].Identity,
+        "Users" : PolicyProtectedUsers,
+        "Action" : Policy[0].TargetedUserProtectionAction
+    }
+} if {
+    Policy := InpersonationProtectionSetting(Policies, IdentityString)
+    count(Policy[0]) > 0
+
+    PolicyProtectedUsers := { x | x := Policy[0].TargetedUsersToProtect[_] }
+    count(PolicyProtectedUsers) > 0
+    count(IncludedUsers - PolicyProtectedUsers) == 0
```

Comment:
_Optional_

I know we decided not to have this check be case-insensitive for 1.4, but we might want it here, because of some weirdness with how Microsoft handles things.

When you go to add the user in the admin center, it is displayed with capitals. 
![image](https://github.com/cisagov/ScubaGear/assets/106177711/bbefc0ea-8b2a-43bb-9d39-f08e55b9609e)

But then the version included in the PowerShell output has been lowercased:
![image](https://github.com/cisagov/ScubaGear/assets/106177711/0f9d2460-f1e6-4fe6-ae72-6b29a67ecad0)

This is contrast to 1.4 where the PowerShell command preserved the case :expressionless:

This of course open to discussion, but I could see this inconsistency with how Microsoft does things confusing our users.

Reply:
So I found a function that turns all letters into lower case, I believe this will allow for it to ignore case. Fix pushed, please confirm